### DISCUSSION TITLE
Gigabit Ethernet emulation

### DISCUSSION THREAD

**Discussion by Torinde** [joncampbell123/dosbox-x#4698] [General]
Intel Pro/1000 MT Desktop (82540EM) - E1000 - is emulated by VirtualBox, Bochs, QEMU and has a [DOS driver](http://packetdriversdos.net), works under Win9x and [probably Win3x](https://www.vogons.org/viewtopic.php?t=81304).

Is it useful to emulate that or even faster cards having DOS drivers?
- REALTEK RTL8125 PCI-E FAMILY 2.5 GIGABIT Ethernet
- Broadcom NetXtreme II 10-GbE

**Comment by rderooy** [comment]
Other then emulating another card type, what is the advantage? The current NE2000 is broadly supported by legacy DOS networking software. The only minor issue is that it does not support PnP, so you need to manually set resources, even when using Win9x. This is where emulating a ISA PnP or PCI adapter could help somewhat, but it would have to be an adapter that was supported by Win9x (preferably out-of-the-box).

Emulating a faster network adapter will not increase speed, as the speed is not being throttled anyway by the emulation.

**Comment by joncampbell123** [comment]
The easiest trick would be to first figure out if there are NE2000 network cards that are ISA Plug & Play or if there exists NE2000 on a PCI card. Then tie in addition code around the NE2000 emulation to participate in the ISA PnP and PCI subsystems of DOSBox-X so that Windows 95 can automatically see it.

**Reply by sskras** [reply]
Looks like there was [a bunch of PCI NE2000 cards](https://webhome.phy.duke.edu/~rgb/brahma/Resources/beowulf/linux/drivers/ne2k-pci.html) manufactured:

> ## Using PCI NE2000 Cards with Linux
> This page contains information on using Linux with PCI NE2000 clones.

[...]

> The following chips are supported by this driver.
> 
> - Compex RL2000
> - Holtek HT80229 and HT80232
> - KTI ET32P2
> - NetVin NV5000SC (really a renumbered RTL8029)
> - RealTek RTL8029 and RTL8029AS
> - Via VT86C926 "Amazon"
> - Winbond 89C940
> 
> The 2.0.35 and later kernels include this driver in the kernel source distribution.
> 
> The similarly numbered [RealTek RTL8139](http://cesdis.gsfc.nasa.gov/linux/drivers/rtl8139.html), [, and ](http://cesdis.gsfc.nasa.gov/linux/drivers/via-rhine.html)[Winbond 89c840](http://cesdis.gsfc.nasa.gov/linux/drivers/ethercard.html) chips are unrelated the chips above, and each has its own driver. All of these unrelated chips are 10/100Mbps designs, while all of the chips supported by the **ne2k-pci** driver are 10Mbps only.

Just my €.02 :)

**Reply by joncampbell123** [reply]
Nice! Alright, pick one, grab it's ISA PnP or PCI device IDs, and let's use to add support!

I expect the best source for the ISA PnP IDs might be the various driver INF files in a typical Windows 95/98/ME/2000/XP install.

**Reply by Torinde** [reply]
At https://www.vogons.org/viewtopic.php?f=46&t=73613 they show some PnP IDs and INF files, is that enough or something has to be checked with actual hardware card?

**Reply by Torinde** [reply]
86Box supports "[NE2000 compatible](https://github.com/86Box/86Box/blob/master/src/network/net_ne2000.c)" PnP devices:
 - Realtek RTL8019AS (ISA 16-bit, PnP);
 - Realtek RTL8029AS (PCI).

**Comment by rderooy** [comment]
related: https://github.com/joncampbell123/dosbox-x/issues/1482
